db/migrations/20221228182055_wrapped.sql

```diff
@@ -0,0 +1,24 @@
+-- migrate:up
+create materialized view user_engagement_wrapped as
+select user_id, sum(total_seconds) total_seconds
+from (
+	SELECT raw.event ->> 'userId' user_id
+	, get_post_id_from_path(raw.event ->> 'path'::text) post_id
+    -- Not all of the data is actually integers :(
+	, max(
+		case when (raw.event ->> 'seconds'::text) similar to '\d+' 
+            then (raw.event ->> 'seconds'::text)::integer
+		else 0
+		end
+	) total_seconds
+	FROM public.raw
+	where event_type = 'timerEvent'
+		and timestamp between '2022-01-01' and '2023-01-01'
+	group by user_id, post_id
```

Comment:
Why are we grouping by `post_id`? Is `timerEvent` related to posts?

Reply:
E.g. the timer goes off after 10 seconds, 30 seconds, 60 seconds, etc. We just want the highest timer.

Note that this doesn't account for if they genuinely view the same post multiple times. In my defense, none of our statistics account for this though.

Reply:
So this doesn't include their time spent on non-post pages, such as the frontpage? Maybe I should update the user-facing text to make that clearer